Unittest fail on big-endian architectures

It seems that the test ` test_get_array_hashable` fails on big endian architectures.
You can find all details about the conditions under which the test has been run at [1].

We are trying to include the new pyresample v1.7.1 in debian but this issue is blocking for us.
Can you pleas provide some hint about a possible solution/workaround?

[1] https://buildd.debian.org/status/package.php?p=pyresample

```
======================================================================
FAIL: test_get_array_hashable (pyresample.test.test_geometry.Test)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/<<PKGBUILDDIR>>/pyresample/test/test_geometry.py", line 226, in test_get_array_hashable
    geometry.get_array_hashable(arr))
  File "/usr/lib/python2.7/dist-packages/numpy/testing/utils.py", line 1395, in assert_allclose
    verbose=verbose, header=header, equal_nan=equal_nan)
  File "/usr/lib/python2.7/dist-packages/numpy/testing/utils.py", line 778, in assert_array_compare
    raise AssertionError(msg)
AssertionError: 
Not equal to tolerance rtol=1e-07, atol=0

(mismatch 50.0%)
 x: array([ 51,  51,  51,  51,  51,  51, 243,  63, 205, 204, 204, 204, 204,
       204, 244,  63, 102, 102, 102, 102, 102, 102, 246,  63,   0,   0,
         0,   0,   0,   0, 248,  63], dtype=uint8)
 y: array([ 63, 243,  51,  51,  51,  51,  51,  51,  63, 244, 204, 204, 204,
       204, 204, 205,  63, 246, 102, 102, 102, 102, 102, 102,  63, 248,
         0,   0,   0,   0,   0,   0], dtype=uint8)

----------------------------------------------------------------------
Ran 158 tests in 851.688s

FAILED (failures=1)
```

--- a/pyresample/test/test_geometry.py
+++ b/pyresample/test/test_geometry.py
@@ -216,13 +216,24 @@ class Test(unittest.TestCase):
 
     def test_get_array_hashable(self):
         arr = np.array([1.2, 1.3, 1.4, 1.5])
+        if sys.byteorder == 'little':
+            # arr.view(np.uint8)
+            reference = np.array([ 51,  51,  51,  51,  51,  51, 243,
+                                   63, 205, 204, 204, 204, 204,
+                                   204, 244,  63, 102, 102, 102, 102,
+                                   102, 102, 246,  63,   0,   0,
+                                   0,   0,   0,   0, 248,  63],
+                                   dtype=np.uint8)
+        else:
+            # on le machines use arr.byteswap().view(np.uint8)
+            reference = np.array([ 63, 243,  51,  51,  51,  51,  51,
+                                   51,  63, 244, 204, 204, 204,
+                                   204, 204, 205,  63, 246, 102, 102,
+                                   102, 102, 102, 102,  63, 248,
+                                   0,   0,   0,   0,   0,   0],
+                                   dtype=np.uint8)
 
-        np.testing.assert_allclose(np.array([ 51,  51,  51,  51,  51,  51, 243,
-                                           63, 205, 204, 204, 204, 204,
-                                           204, 244,  63, 102, 102, 102, 102,
-                                           102, 102, 246,  63,   0,   0,
-                                           0,   0,   0,   0, 248,  63],
-                                           dtype=np.uint8),
+        np.testing.assert_allclose(reference,
                                     geometry.get_array_hashable(arr))
 
         try:
@@ -231,13 +242,8 @@ class Test(unittest.TestCase):
             pass
         else:
             xrarr = xr.DataArray(arr)
-            np.testing.assert_allclose(np.array([ 51,  51,  51,  51,  51,  51, 243,
-                                           63, 205, 204, 204, 204, 204,
-                                           204, 244,  63, 102, 102, 102, 102,
-                                           102, 102, 246,  63,   0,   0,
-                                           0,   0,   0,   0, 248,  63],
-                                           dtype=np.uint8),
-                                    geometry.get_array_hashable(arr))
+            np.testing.assert_allclose(reference,
+                                       geometry.get_array_hashable(arr))
 
             xrarr.attrs['hash'] = 42
             self.assertEqual(geometry.get_array_hashable(xrarr),

